django/utils/timezone.py

```diff
@@ -295,8 +300,13 @@ def make_naive(value, timezone=None):
     """
     if timezone is None:
         timezone = get_current_timezone()
-    # If `value` is naive, astimezone() will raise a ValueError,
-    # so we don't need to perform a redundant check.
+
+    # astimezone() no longer raises a ValueError on Python 3.6+.
+    # Check it explicitly for consistency across Python versions.
+    if is_naive(value):
+        raise ValueError(
+            "make_naive() cannot be applied to a naive datetime")
+
```

Comment:
I was discussing with @aaugustin who told me it's not necessary to put those 2 copy/pasted snippets in a single function